interpreter/src/main/scala/pl/touk/nussknacker/engine/types/EspTypeUtils.scala

```diff
@@ -114,16 +121,63 @@ object EspTypeUtils {
   }
 
   private def extractMethod(method: Method)
-    = MethodInfo(extractParameters(method), extractMethodReturnType(method), extractNussknackerDocs(method), method.isVarArgs)
+                           (implicit settings: ClassExtractionSettings): List[MethodInfo] =
+    method.getAnnotation(classOf[GenericType]) match {
+      case null => extractRegularMethod(method)
+      case annotation =>
+        try {
+          extractGenericMethod(method, annotation)
+        } catch {
+          case e: InstantiationException =>
```

Comment:
or just case NonFatal(e) => throw new IllegalArgumentException("Could not create TypingFunction", e)
And please, catch it earlier - in extractGenericMethod?